projects/packages/connection-ui/_inc/reducers/connection-status.js

```diff
@@ -1,23 +1,12 @@
 /**
  * Internal dependencies
  */
-import {
-	CONNECTION_STATUS_REGISTERED,
-	CONNECTION_STATUS_USER_CONNECTED,
-} from '../actions/connection-status';
+import { SET_CONNECTION_STATUS } from '../actions/connection-status';
 
 const connectionStatus = ( state = {}, action ) => {
 	switch ( action.type ) {
-		case CONNECTION_STATUS_REGISTERED:
-			return {
-				...state,
-				isRegistered: action.isRegistered,
-			};
-		case CONNECTION_STATUS_USER_CONNECTED:
-			return {
-				...state,
-				isUserConnected: action.isUserConnected,
-			};
+		case SET_CONNECTION_STATUS:
+			return action.connectionStatus;
```

Comment:
Is there a reason to completely overwrite the state object here versus using:
`return { ...state, ...action.connectionStatus };` ?

Reply:
I believe this makes the behavior more straightforward.
The spread construction you're suggesting is elegant, but will only merge top-level elements, while sub-object elements (e.g. children of `offlineMode`) will get overwritten.

So to get rid of such edge cases I decided to overwrite the object completely in the same way it's received from the RNA Connection. If there's a need to store some custom connection data, they should be stored separately.